tests/bluetooth/df/connection_cte_req/src/test_cte_set_rx_params.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stddef.h>
+#include <ztest.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <sys/byteorder.h>
+#include <host/hci_core.h>
+
+#include "common.h"
+#include "test_cte_set_rx_params.h"
+
+static uint16_t g_conn_handle;
+
+static uint8_t g_ant_ids[] = { 0x1, 0x2, 0x3, 0x4, 0x5 };
+
+static struct ut_bt_df_conn_cte_rx_params g_params;
+
+/* Macros delivering common values for unit tests */
+#define CONN_HANDLE_INVALID (CONFIG_BT_MAX_CONN + 1)
+#define ANTENNA_SWITCHING_SLOT_INVALID 0x3 /* BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US + 1 */
+
+/* Antenna switch pattern length is stored in 1 octet. If BT Core spec. extends the max value to
+ * UINT8_MAX expected failures may not be checked. If storage size is increased, tests shall be
+ * updated.
+ */
+BUILD_ASSERT(CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN < UINT8_MAX, "Can't test expected failures "
+								   "for switch pattern length "
+								   "longer or equal to UINT8_MAX.");
+#define SWITCH_PATTERN_LEN_TOO_LONG (CONFIG_BT_CTLR_DF_MAX_ANT_SW_PATTERN_LEN + 1)
+
+BUILD_ASSERT(BT_HCI_LE_SWITCH_PATTERN_LEN_MIN > 0x0, "Can't test expected failures for switch "
+						     "pattern length smaller than zero.");
```

Comment:
```suggestion
BUILD_ASSERT(BT_HCI_LE_SWITCH_PATTERN_LEN_MIN > 0x0, 
	     "Can't test expected failures for switch pattern length smaller "
	     "than zero.");
```

Is it possible format strings for BUILD_ASSERT? If so, it might be worth also writing the value of the variable that is a problem, e.g.:

```suggestion
BUILD_ASSERT(BT_HCI_LE_SWITCH_PATTERN_LEN_MIN > 0x0, 
	     "Can't test expected failures for switch pattern length smaller (%d) "
	     "than zero.", BT_HCI_LE_SWITCH_PATTERN_LEN_MIN);
```

Reply:
According to documentation: 
>Unlike __ASSERT(), the message must be a static string, without printf()-like format codes or extra arguments.
I've extended the message to inform than switch pattern length may not be smaller or equal to zero.